Microsoft.Toolkit.Uwp.UI.Controls/TabView/TabView.cs

```diff
@@ -0,0 +1,395 @@
+        protected override Size MeasureOverride(Size availableSize)
+        {
+            return base.MeasureOverride(availableSize);
+        }
+
+        /// <inheritdoc/>
+        protected override void OnApplyTemplate()
+        {
+            base.OnApplyTemplate();
+
+            _tabContentPresenter = GetTemplateChild(TABCONTENTPRESENTER_NAME) as ContentPresenter;
+            _tabViewContainer = GetTemplateChild(TABVIEWCONTAINER_NAME) as Grid;
+            _tabItemsPresenter = GetTemplateChild(TABITEMSPRESENTER_NAME) as ItemsPresenter;
+            _tabScroller = GetTemplateChild(TABSCROLLVIEWER_NAME) as ScrollViewer;
+
+            DragLeave += TabPresenter_DragLeave;
+            DragItemsCompleted += TabPresenter_DragItemsCompleted;
+            SizeChanged += TabView_SizeChanged;
+
+            if (_tabItemsPresenter != null)
+            {
+                _tabItemsPresenter.SizeChanged -= TabView_SizeChanged;
+                _tabItemsPresenter.SizeChanged += TabView_SizeChanged;
+            }
+
+            if (_tabContentPresenter != null)
+            {
+                SelectionChanged -= TabView_SelectionChanged;
+                SelectionChanged += TabView_SelectionChanged;
+            }
+
+            if (_tabScroller != null)
+            {
+                _tabScroller.Loaded -= ScrollViewer_Loaded;
+                _tabScroller.Loaded += ScrollViewer_Loaded;
+            }
+        }
+
+        private void ScrollViewer_Loaded(object sender, RoutedEventArgs e)
+        {
+            _tabScroller.Loaded -= ScrollViewer_Loaded;
+
+            _tabScrollBackButton = _tabScroller.FindDescendantByName(TABSCROLLBACKBUTTON_NAME) as ButtonBase;
+            _tabScrollForwardButton = _tabScroller.FindDescendantByName(TABSCROLLFORWARDBUTTON_NAME) as ButtonBase;
+
+            if (_tabScrollBackButton != null)
+            {
+                _tabScrollBackButton.Click -= ScrollTabBackButton_Click;
+                _tabScrollBackButton.Click += ScrollTabBackButton_Click;
+            }
+
+            if (_tabScrollForwardButton != null)
+            {
+                _tabScrollForwardButton.Click -= ScrollTabForwardButton_Click;
+                _tabScrollForwardButton.Click += ScrollTabForwardButton_Click;
+            }
+        }
+
+        private void ScrollTabBackButton_Click(object sender, RoutedEventArgs e)
+        {
+            if (_tabScroller != null)
+            {
+                _tabScroller.ChangeView(Math.Max(0, _tabScroller.HorizontalOffset - SCROLL_AMOUNT), null, null);
+            }
+        }
+
+        private void ScrollTabForwardButton_Click(object sender, RoutedEventArgs e)
+        {
+            if (_tabScroller != null)
+            {
+                _tabScroller.ChangeView(Math.Min(_tabScroller.ScrollableWidth, _tabScroller.HorizontalOffset + SCROLL_AMOUNT), null, null);
+            }
+        }
+
+        private void TabView_SizeChanged(object sender, SizeChangedEventArgs e)
+        {
```

Comment:
I think the tabs should shrink to a min width as the TabView gets smaller. Just as Edge does.

Reply:
That's what the EdgeTabWidthProvider does.